internal/controller/controller_test.go

```diff
@@ -503,7 +503,106 @@ func TestHttpController_ExportRecordedData(t *testing.T) {
 }
 
 func TestHttpController_ImportRecordedData(t *testing.T) {
-	// TODO: Implement using TDD
+	target, mockDataManager, _ := createTargetAndMocks()
+
+	handler := http.HandlerFunc(target.importRecordedData)
+
+	emptyDataRequest := dtos.RecordedData{}
+
+	recordedEventRequest := dtos.RecordedData{
+		RecordedEvents: []coreDtos.Event{
+			coreDtos.Event{
+				DeviceName:  "test",
+				ProfileName: "test",
+				Readings: []coreDtos.BaseReading{
+					coreDtos.BaseReading{
+						SimpleReading: coreDtos.SimpleReading{
+							Value: "1456.0",
+						},
+					},
+				},
+			},
+			coreDtos.Event{
+				DeviceName:  "test",
+				ProfileName: "test",
+				Readings: []coreDtos.BaseReading{
+					coreDtos.BaseReading{
+						SimpleReading: coreDtos.SimpleReading{
+							Value: "1457.0",
+						},
+					},
+				},
+			},
+		},
+		Devices: []coreDtos.Device{
+			coreDtos.Device{
+				Name:        "test_device",
+				ProfileName: "test",
+			},
+		},
+	}
+
+	tests := []struct {
+		Name             string
+		CompressionParam string
+		OverwriteParam   string
+		ExpectedResponse []byte
+		ExpectedStatus   int
+		ExpectedError    error
+	}{
+		{
+			Name:             "valid - data with 2 events",
+			ExpectedResponse: marshal(t, recordedEventRequest),
+			ExpectedStatus:   http.StatusAccepted,
+			ExpectedError:    nil,
+			OverwriteParam:   "false",
+		},
+		{
+			Name:             "invalid - no data",
+			ExpectedResponse: marshal(t, emptyDataRequest),
+			ExpectedStatus:   http.StatusNoContent,
+			ExpectedError:    errors.New(noDataFound),
+			OverwriteParam:   "false",
+		},
+		{
+			Name:             "valid - data with 2 events and compressed gzip",
+			CompressionParam: "GZIP",
+			ExpectedResponse: compressData(t, "GZIP", recordedEventRequest),
+			ExpectedStatus:   http.StatusAccepted,
+			ExpectedError:    nil,
+			OverwriteParam:   "false",
+		},
+		{
+			Name:             "valid - data with 2 events and compressed zlib",
+			CompressionParam: "ZLIB",
+			ExpectedResponse: compressData(t, "ZLIB", recordedEventRequest),
+			ExpectedStatus:   http.StatusAccepted,
+			ExpectedError:    nil,
+			OverwriteParam:   "false",
+		},
+	}
+	for _, test := range tests {
+		t.Run(test.Name, func(t *testing.T) {
+			// mockDataManager will not be called if no data is sent
+			if test.ExpectedStatus != http.StatusNoContent {
+				mockDataManager.On("ImportRecordedData", mock.Anything, mock.Anything).Return(test.ExpectedError).Once()
+			}
+
+			req, err := http.NewRequest(http.MethodPost, dataRoute, bytes.NewReader(test.ExpectedResponse))
+			require.NoError(t, err)
+
+			query := req.URL.Query()
+			query.Add("compression", test.CompressionParam)
```

Comment:
Need to set `Content-Encoding` instead using standard HTTP header values.
https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Encoding